microsoftgraph/msgraph-metadata
# Add Custom Query Parameters To CalendarView Nav Property

This PR fixes https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/64 issue by adding [CustomQueryOptions](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Capabilities.V1.xml#L915) capability annotation to `users`, `groups` and `bookingBusinesses` `EntitySets`. 

Custom Properties Added:
| [Users/CalendarView](https://docs.microsoft.com/en-us/graph/api/user-list-calendarview?view=graph-rest-beta&tabs=http) | [Groups/CalendarView](https://docs.microsoft.com/en-us/graph/api/group-list-calendarview?view=graph-rest-beta&tabs=http) | [BookingBusinesses/CalendarView](https://docs.microsoft.com/en-us/graph/api/bookingbusiness-list-calendarview?view=graph-rest-beta&tabs=http)  |
| ------------- |:-------------:| -----:|
| startDateTime | startDateTime | start |
| endDateTime | endDateTime | end |


To test:
1. Open Visual Studio without a project.
2. Open `preprocess_csdl_test_input.xml`.
3. Select the **XML** menu from the top menu.
4. Select the **Start XSLT Without Debugging** option.

The generated file should match preprocess_csdl_test_output.xml.
The generated file should not match preprocess_csdl_test_input.xml.
The corresponding OpenAPI docs will then have the custom query options - `startDateTime` & `endDateTime`:
![image](https://user-images.githubusercontent.com/7061532/80153414-4f371f80-8572-11ea-80a1-1a80c7cf67aa.png)
